src/sidebar/util/test/session-util-test.js

```diff
@@ -2,24 +2,94 @@
 
 const sessionUtil = require('../session-util');
 
-describe('sessionUtil.shouldShowSidebarTutorial', function() {
-  it('shows sidebar tutorial if the settings object has the show_sidebar_tutorial key set', function() {
-    const sessionState = {
-      preferences: {
-        show_sidebar_tutorial: true,
-      },
-    };
+describe('sidebar.util.session-util', () => {
+  describe('#shouldShowSidebarTutorial', () => {
+    it('shows sidebar tutorial if the settings object has the show_sidebar_tutorial key set', function() {
+      const sessionState = {
+        preferences: {
+          show_sidebar_tutorial: true,
+        },
+      };
+
+      assert.isTrue(sessionUtil.shouldShowSidebarTutorial(sessionState));
+    });
+
+    it('hides sidebar tutorial if the settings object does not have the show_sidebar_tutorial key set', function() {
+      const sessionState = {
+        preferences: {
+          show_sidebar_tutorial: false,
+        },
+      };
 
-    assert.isTrue(sessionUtil.shouldShowSidebarTutorial(sessionState));
+      assert.isFalse(sessionUtil.shouldShowSidebarTutorial(sessionState));
+    });
   });
 
-  it('hides sidebar tutorial if the settings object does not have the show_sidebar_tutorial key set', function() {
-    const sessionState = {
-      preferences: {
-        show_sidebar_tutorial: false,
+  describe('#shouldAutoDisplayTutorial', () => {
+    [
+      {
+        // The only "true" state
+        description: 'in sidebar with loaded user preference to show tutorial',
+        isSidebar: true,
+        sessionState: { preferences: { show_sidebar_tutorial: true } },
+        settings: {},
+        expected: true,
       },
-    };
-
-    assert.isFalse(sessionUtil.shouldShowSidebarTutorial(sessionState));
+      {
+        description: 'in sidebar with no loaded user profile',
+        isSidebar: true,
+        sessionState: {},
+        settings: {},
+        expected: false,
+      },
+      {
+        description: 'not in sidebar with no loaded user profile',
+        isSidebar: false,
+        sessionState: {},
+        settings: {},
+        expected: false,
+      },
+      {
+        description:
+          'in sidebar with loaded user preference not to show tutorial',
+        isSidebar: true,
+        sessionState: { preferences: { show_sidebar_tutorial: false } },
+        settings: {},
+        expected: false,
+      },
+      {
+        description:
+          'in sidebar with loaded user preference to show tutorial and configured help service',
+        isSidebar: true,
+        sessionState: { preferences: { show_sidebar_tutorial: true } },
+        settings: { services: [{ onHelpRequestProvided: true }] },
```

Comment:
Careful eyes here, please. As I understand it, by the time `settings` have made it to this point in the app lifecycle, they've been parsed/munged such that the incoming `settings.services[0]` object's `onHelpRequest` (function reference) is converted to `onHelpRequestProvided: true`. This seems to be the case if I mess with my local `live-reload` config object in such a way, but it's a little convoluted so I want some attention here to make sure I got it right.